### PR TITLE
Update ECMWF odc library, add to jedi-base-env

### PR DESCRIPTION
**Pull request by srherbener**
This PR upgrades the odc packaging to have access to newer versions (version 1.4.5 in particular).

This PR needs to be merged with noaa-emc/spack-stack/pull/283

### CODE REVIEW

**Review by climbfuji — COMMENTED** [review]
This looks good, but we should test this PR in the spack-stack CI tests and probably also one of the HPCs.

**Comment by climbfuji** [comment]
I see that the CI tests are already running, great. We'll need to test on HPCs (manually for now). I can take care of this.

File: var/spack/repos/builtin/packages/odc/package.py [hunk 1]
```diff
@@ -14,6 +14,10 @@ class Odc(CMakePackage):
 
     maintainers = ['skosukhin']
 
+    version('1.4.5', sha256='8532d0453531d62e1f15791d1c5c96540b842913bd211a8ef090211eaf4cccae', preferred=True)
```
Comment (kgerheiser) [inline var/spack/repos/builtin/packages/odc/package.py]: Is `preferred=True` necessary here? The latest version is always preferred, unless there's some other reason that 1.4.5 should be preferred.

**Comment by climbfuji** [comment]
Also, we have used the convention to put packages that are not yet in the authoritative spack repo in the repo `jcsda_emc` instead of `builtin`. This will make it easier for us to know what needs to be send to upstream etc.

File: var/spack/repos/builtin/packages/odc/package.py [hunk 2]
```diff
@@ -14,6 +14,10 @@ class Odc(CMakePackage):
 
     maintainers = ['skosukhin']
```
Comment (climbfuji) [inline var/spack/repos/builtin/packages/odc/package.py:15]: I think this should be @srherbener and @climbfuji for now

**Comment by kgerheiser** [comment]
@climbfuji this is already a `builtin` package

**Comment by climbfuji** [comment]
Argh. Sorry.

**Comment by climbfuji** [comment]
> Argh. Sorry.
> […](#)

I was confused by the title of the PR ("Add ECMWF odc library") - should be "Update ..." ?

Also, in this case, please leave the maintainer as is.

File: var/spack/repos/builtin/packages/odc/package.py (hunk 1 above)
Comment (srherbener) [reply var/spack/repos/builtin/packages/odc/package.py]: I'm okay with removing the `preferred=True`. I don't think there is anything special about 1.4.5 beside it being the latest available version. I'll go ahead and make that change.

**Comment by srherbener** [comment]
> Also, we have used the convention to put packages that are not yet in the authoritative spack repo in the repo `jcsda_emc` instead of `builtin`. This will make it easier for us to know what needs to be send to upstream etc.

The odc package was already in existence (in the `builtin` repo) so I used that. I'm okay moving it to `jcsda_emc` if it's better to place it there.

**Comment by climbfuji** [comment]
All good, I updated the title of the PR and the package can stay where it is. I was just confused.

**Review by climbfuji — APPROVED** [review]
I built the latest version on macOS, on Cheyenne with Intel, and on Cheyenne with GNU. Good to merge!

**Review by kgerheiser — APPROVED** [review]

**Comment by climbfuji** [comment]
@srherbener spack PR is merged, new hash is https://github.com/NOAA-EMC/spack/commit/82e413672a9722caf4781a9df49493b855d46850 - can you please update your spack-stack PR?

**Comment by srherbener** [comment]
> @srherbener spack PR is merged, new hash is [82e4136](https://github.com/NOAA-EMC/spack/commit/82e413672a9722caf4781a9df49493b855d46850) - can you please update your spack-stack PR?

Done!

**Pull request merged**